### PR TITLE
[yang]: Add Yang models for BGP monitor

### LINKED ISSUE #9711: [yang-models] missing yang model for BGP_MONITORS
#### Description
[Generic Config Updater](https://github.com/Azure/SONiC/blob/master/doc/config-generic-update-rollback/SONiC_Generic_Config_Update_and_Rollback_Design.md#91-unit-tests-for-apply-patch) has a test case about BGP listener which refers to the configDB table BGP_MONITORS

#### Steps to reproduce the issue:
1. `apply-patch` to add config for BGP_MONITORS
2. action fails
3.

#### Describe the results you received:
```
admin@vlab-01:~$ sudo config apply-patch -i /FEATURE -i /QUEUE -i /SCHEDULER bgpl/add.json
Patch Applier: Patch application starting.
Patch Applier: Patch: [{"op": "add", "path": "/BGP_MONITORS", "value": {"11.0.0.1": {"admin_status": "up", "asn": "65100", "holdtime": "10", "keepalive": "3", "local_addr": "10.1.0.32", "name": "bgp_monitor", "nhopself": "0", "rrclient": "0"}}}]
Patch Applier: Getting current config db.
Patch Applier: Simulating the target full config after applying the patch.
Patch Applier: Validating target config does not have empty tables, since they do not show up in ConfigDb.
Patch Applier: Sorting patch updates.
Note: Below table(s) have no YANG models:
BGP_PEER_RANGE, CONSOLE_SWITCH, DEVICE_NEIGHBOR_METADATA, DHCP_SERVER, KDUMP, RESTAPI, SNMP, SNMP_COMMUNITY, TELEMETRY, BGP_MONITORS,
Note: Below table(s) have no YANG models:
BGP_MONITORS,
Failed to apply patch

```

#### Describe the results you expected:
`apply-patch` should not report miss Yang model

#### Output of `show version`:

```
admin@vlab-01:~/$ show version

SONiC Software Version: SONiC.master.58770-6402a0226
Distribution: Debian 11.1
Kernel: 5.10.0-8-2-amd64
Build commit: 6402a0226
Build date: Mon Dec 13 09:02:14 UTC 2021
Built by: AzDevOps@sonic-build-workers-000Z9S

Platform: x86_64-kvm_x86_64-r0
HwSKU: Force10-S6000
ASIC: vs
ASIC Count: 1
Serial Number: N/A
Model Number: N/A
Hardware Revision: N/A
Uptime: 07:01:22 up 6 days,  3:30,  1 user,  load average: 0.19, 0.30, 0.45
```
#### Output of `show techsupport`:

```
(paste your output here or download and attach the file here )
```

### PR DESCRIPTION
**Pull request by ganglyu**
Signed-off-by: Gang Lv ganglv@microsoft.com

#### Why I did it
end2end test is blocked by Yang model for BGP monitor.

#### How I did it
Create new yang files for BGP monitor, and add UT.

#### How to verify it
Follow the steps in #9711.
Run UT for sonic-yang-models.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106

#### Description for the changelog

Fix #9711

### CODE REVIEW

**Comment by wen587** [comment]
Tested on KVM env. `apply-patch` can succeed. But bgpcfgd report ERR. It may be caused by invalid value in my patch because I am not sure how to configure that. 

BTW, I saw in production that `asn` could be 0.
```
admin@vlab-01:~/bgpl$ cat add.json
[ {
  "op": "add",
  "path": "/BGP_MONITORS",
  "value": {
   "10.0.0.57": {
    "asn": "1",
    "holdtime": "180",
    "keepalive": "60",
    "local_addr": "10.0.0.58",
    "name": "BGPMonitor",
    "nhopself": "0",
    "rrclient": "0"
   }
  }
 }
]

admin@vlab-01:~/bgpl$ sudo cat /var/log/syslog | less

Jan 13 07:12:17.930661 vlab-01 NOTICE config: Patch Applier: Applying 7 changes in order:
Jan 13 07:12:17.930783 vlab-01 NOTICE config: Patch Applier:   * [{"op": "add", "path": "/BGP_MONITORS", "value": {"10.0.0.57": {"asn": "1"}}}]
Jan 13 07:12:18.269504 vlab-01 DEBUG bgp#bgpcfgd: Received message : '('10.0.0.57', 'SET', (('asn', '1'),))'
Jan 13 07:12:18.271356 vlab-01 WARNING bgp#bgpcfgd: Peer 10.0.0.57. Missing attribute 'local_addr'
Jan 13 07:12:18.272549 vlab-01 INFO bgp#bgpcfgd: Routing policy for peer '10.0.0.57' has been scheduled to be updated
Jan 13 07:12:18.272549 vlab-01 INFO bgp#bgpcfgd: Peer-group for peer '10.0.0.57' has been scheduled to be updated
Jan 13 07:12:18.273635 vlab-01 INFO bgp#bgpcfgd: Peer '(default|10.0.0.57)' has been scheduled to be added with attributes '{'asn': '1'}'
Jan 13 07:12:18.274026 vlab-01 DEBUG bgp#bgpcfgd: execute command '['vtysh', '-f', '/tmp/tmph_geju49']'.
Jan 13 07:12:18.399504 vlab-01 ERR bgp#bgpcfgd: command execution returned 2. Command: '['vtysh', '-f', '/tmp/tmph_geju49']', stdout: '', stderr: 'line 29: % Unknown command[24]:   neighbor 10.0.0.57 description #012'
Jan 13 07:12:18.400561 vlab-01 ERR bgp#bgpcfgd: ConfigMgr::commit(): can't push configuration from file='/tmp/tmph_geju49', rc='2', stdout='', stderr='line 29: % Unknown command[24]:   neighbor 10.0.0.57 description #012'
Jan 13 07:12:18.401146 vlab-01 CRIT bgp#bgpcfgd: Runner::commit was unsuccessful
Jan 13 07:12:18.574125 vlab-01 NOTICE config: Patch Applier:   * [{"op": "add", "path": "/BGP_MONITORS/10.0.0.57/holdtime", "value": "180"}]
Jan 13 07:12:18.926387 vlab-01 DEBUG bgp#bgpcfgd: Received message : '('10.0.0.57', 'SET', (('asn', '1'), ('holdtime', '180')))'
Jan 13 07:12:18.926801 vlab-01 ERR bgp#bgpcfgd: Peer '(default|10.0.0.57)': Can't update the peer. Only 'admin_status' attribute is supported
Jan 13 07:12:19.269490 vlab-01 NOTICE config: Patch Applier:   * [{"op": "add", "path": "/BGP_MONITORS/10.0.0.57/keepalive", "value": "60"}]
Jan 13 07:12:19.599301 vlab-01 DEBUG bgp#bgpcfgd: Received message : '('10.0.0.57', 'SET', (('asn', '1'), ('holdtime', '180'), ('keepalive', '60')))'
Jan 13 07:12:19.599526 vlab-01 ERR bgp#bgpcfgd: Peer '(default|10.0.0.57)': Can't update the peer. Only 'admin_status' attribute is supported
Jan 13 07:12:19.926194 vlab-01 NOTICE config: Patch Applier:   * [{"op": "add", "path": "/BGP_MONITORS/10.0.0.57/local_addr", "value": "10.0.0.58"}]
Jan 13 07:12:20.264585 vlab-01 DEBUG bgp#bgpcfgd: Received message : '('10.0.0.57', 'SET', (('asn', '1'), ('holdtime', '180'), ('keepalive', '60'), ('local_addr', '10.0.0.58')))'
Jan 13 07:12:20.264926 vlab-01 ERR bgp#bgpcfgd: Peer '(default|10.0.0.57)': Can't update the peer. Only 'admin_status' attribute is supported
Jan 13 07:12:20.574737 vlab-01 NOTICE config: Patch Applier:   * [{"op": "add", "path": "/BGP_MONITORS/10.0.0.57/name", "value": "BGPMonitor"}]
Jan 13 07:12:20.896012 vlab-01 DEBUG bgp#bgpcfgd: Received message : '('10.0.0.57', 'SET', (('asn', '1'), ('holdtime', '180'), ('keepalive', '60'), ('local_addr', '10.0.0.58'), ('name', 'BGPMonitor')))'
Jan 13 07:12:20.896012 vlab-01 ERR bgp#bgpcfgd: Peer '(default|10.0.0.57)': Can't update the peer. Only 'admin_status' attribute is supported
Jan 13 07:12:21.228961 vlab-01 NOTICE config: Patch Applier:   * [{"op": "add", "path": "/BGP_MONITORS/10.0.0.57/nhopself", "value": "0"}]
Jan 13 07:12:21.562566 vlab-01 DEBUG bgp#bgpcfgd: Received message : '('10.0.0.57', 'SET', (('asn', '1'), ('holdtime', '180'), ('keepalive', '60'), ('local_addr', '10.0.0.58'), ('name', 'BGPMonitor'), ('nhopself', '0')))'
Jan 13 07:12:21.562566 vlab-01 ERR bgp#bgpcfgd: Peer '(default|10.0.0.57)': Can't update the peer. Only 'admin_status' attribute is supported
Jan 13 07:12:21.898524 vlab-01 NOTICE config: Patch Applier:   * [{"op": "add", "path": "/BGP_MONITORS/10.0.0.57/rrclient", "value": "0"}]
Jan 13 07:12:22.256744 vlab-01 DEBUG bgp#bgpcfgd: Received message : '('10.0.0.57', 'SET', (('asn', '1'), ('holdtime', '180'), ('keepalive', '60'), ('local_addr', '10.0.0.58'), ('name', 'BGPMonitor'), ('nhopself', '0'), ('rrclient', '0')))'
Jan 13 07:12:22.256744 vlab-01 ERR bgp#bgpcfgd: Peer '(default|10.0.0.57)': Can't update the peer. Only 'admin_status' attribute is supported
Jan 13 07:12:22.625071 vlab-01 NOTICE config: Patch Applier: Verifying patch updates are reflected on ConfigDB.
Jan 13 07:12:23.670024 vlab-01 NOTICE config: Patch Applier: Patch application completed.


```

File: src/sonic-yang-models/yang-models/sonic-bgp-monitor.yang [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+                    }
+                    description "Nexthop is self, no nexthop calculation";
+                }
+
+                leaf rrclient {
+                    type uint8 {
+                        range "0..1";
+                    }
+                    description "Route reflector client";
+                }
```
Comment (shi-su) [inline src/sonic-yang-models/yang-models/sonic-bgp-monitor.yang:81]: I am not sure if `nhopself ` and `rrclient` are required fields. Could you please confirm?

**Comment by qiluo-msft** [comment]
@venkatmahalingam Could you help review?

**Comment by venkatmahalingam** [comment]
> @venkatmahalingam Could you help review?

Sure, will do.

File: src/sonic-yang-models/yang-models/sonic-bgp-monitor.yang [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    description
+        "SONIC BGP Monitor";
+
+    revision 2022-01-11 {
+        description
+            "Initial revision.";
+    }
+
+    container sonic-bgp-monitor {
+        container BGP_MONITORS {
```
Comment (venkatmahalingam) [inline src/sonic-yang-models/yang-models/sonic-bgp-monitor.yang:29]: This table fields seem to be duplicate from the table BGP_NEIGHBOR, please clarify why do we need a separate table, also, please use the grouping to group the fields and use them in multiple places.

File: src/sonic-yang-models/tests/yang_model_tests/tests_config/bgp.json [hunk 3]
```diff
@@ -1322,8 +1322,87 @@
                 ]
             }
         }
-    }
+    },
```
Comment (venkatmahalingam) [inline src/sonic-yang-models/tests/yang_model_tests/tests_config/bgp.json:1326]: I think, we should add YANG for BGP_PEER_RANGE table as well, please check and create the PR for review.

File: src/sonic-yang-models/yang-models/sonic-bgp-monitor.yang (hunk 1 above)
Comment (venkatmahalingam) [reply src/sonic-yang-models/yang-models/sonic-bgp-monitor.yang:81]: Per bgpcfgd code, these are required fields.

**Comment by wen587** [comment]
> Tested on KVM env. `apply-patch` can succeed. But bgpcfgd report ERR. It may be caused by invalid value in my patch because I am not sure how to configure that.
> 
> BTW, I saw in production that `asn` could be 0.
> 
> ```
> admin@vlab-01:~/bgpl$ cat add.json
> [ {
>   "op": "add",
>   "path": "/BGP_MONITORS",
>   "value": {
>    "10.0.0.57": {
>     "asn": "1",
>     "holdtime": "180",
>     "keepalive": "60",
>     "local_addr": "10.0.0.58",
>     "name": "BGPMonitor",
>     "nhopself": "0",
>     "rrclient": "0"
>    }
>   }
>  }
> ]
> 
> admin@vlab-01:~/bgpl$ sudo cat /var/log/syslog | less
> 
> Jan 13 07:12:17.930661 vlab-01 NOTICE config: Patch Applier: Applying 7 changes in order:
> Jan 13 07:12:17.930783 vlab-01 NOTICE config: Patch Applier:   * [{"op": "add", "path": "/BGP_MONITORS", "value": {"10.0.0.57": {"asn": "1"}}}]
> Jan 13 07:12:18.269504 vlab-01 DEBUG bgp#bgpcfgd: Received message : '('10.0.0.57', 'SET', (('asn', '1'),))'
> Jan 13 07:12:18.271356 vlab-01 WARNING bgp#bgpcfgd: Peer 10.0.0.57. Missing attribute 'local_addr'
> Jan 13 07:12:18.272549 vlab-01 INFO bgp#bgpcfgd: Routing policy for peer '10.0.0.57' has been scheduled to be updated
> Jan 13 07:12:18.272549 vlab-01 INFO bgp#bgpcfgd: Peer-group for peer '10.0.0.57' has been scheduled to be updated
> Jan 13 07:12:18.273635 vlab-01 INFO bgp#bgpcfgd: Peer '(default|10.0.0.57)' has been scheduled to be added with attributes '{'asn': '1'}'
> Jan 13 07:12:18.274026 vlab-01 DEBUG bgp#bgpcfgd: execute command '['vtysh', '-f', '/tmp/tmph_geju49']'.
> Jan 13 07:12:18.399504 vlab-01 ERR bgp#bgpcfgd: command execution returned 2. Command: '['vtysh', '-f', '/tmp/tmph_geju49']', stdout: '', stderr: 'line 29: % Unknown command[24]:   neighbor 10.0.0.57 description #012'
> Jan 13 07:12:18.400561 vlab-01 ERR bgp#bgpcfgd: ConfigMgr::commit(): can't push configuration from file='/tmp/tmph_geju49', rc='2', stdout='', stderr='line 29: % Unknown command[24]:   neighbor 10.0.0.57 description #012'
> Jan 13 07:12:18.401146 vlab-01 CRIT bgp#bgpcfgd: Runner::commit was unsuccessful
> Jan 13 07:12:18.574125 vlab-01 NOTICE config: Patch Applier:   * [{"op": "add", "path": "/BGP_MONITORS/10.0.0.57/holdtime", "value": "180"}]
> Jan 13 07:12:18.926387 vlab-01 DEBUG bgp#bgpcfgd: Received message : '('10.0.0.57', 'SET', (('asn', '1'), ('holdtime', '180')))'
> Jan 13 07:12:18.926801 vlab-01 ERR bgp#bgpcfgd: Peer '(default|10.0.0.57)': Can't update the peer. Only 'admin_status' attribute is supported
> Jan 13 07:12:19.269490 vlab-01 NOTICE config: Patch Applier:   * [{"op": "add", "path": "/BGP_MONITORS/10.0.0.57/keepalive", "value": "60"}]
> Jan 13 07:12:19.599301 vlab-01 DEBUG bgp#bgpcfgd: Received message : '('10.0.0.57', 'SET', (('asn', '1'), ('holdtime', '180'), ('keepalive', '60')))'
> Jan 13 07:12:19.599526 vlab-01 ERR bgp#bgpcfgd: Peer '(default|10.0.0.57)': Can't update the peer. Only 'admin_status' attribute is supported
> Jan 13 07:12:19.926194 vlab-01 NOTICE config: Patch Applier:   * [{"op": "add", "path": "/BGP_MONITORS/10.0.0.57/local_addr", "value": "10.0.0.58"}]
> Jan 13 07:12:20.264585 vlab-01 DEBUG bgp#bgpcfgd: Received message : '('10.0.0.57', 'SET', (('asn', '1'), ('holdtime', '180'), ('keepalive', '60'), ('local_addr', '10.0.0.58')))'
> Jan 13 07:12:20.264926 vlab-01 ERR bgp#bgpcfgd: Peer '(default|10.0.0.57)': Can't update the peer. Only 'admin_status' attribute is supported
> Jan 13 07:12:20.574737 vlab-01 NOTICE config: Patch Applier:   * [{"op": "add", "path": "/BGP_MONITORS/10.0.0.57/name", "value": "BGPMonitor"}]
> Jan 13 07:12:20.896012 vlab-01 DEBUG bgp#bgpcfgd: Received message : '('10.0.0.57', 'SET', (('asn', '1'), ('holdtime', '180'), ('keepalive', '60'), ('local_addr', '10.0.0.58'), ('name', 'BGPMonitor')))'
> Jan 13 07:12:20.896012 vlab-01 ERR bgp#bgpcfgd: Peer '(default|10.0.0.57)': Can't update the peer. Only 'admin_status' attribute is supported
> Jan 13 07:12:21.228961 vlab-01 NOTICE config: Patch Applier:   * [{"op": "add", "path": "/BGP_MONITORS/10.0.0.57/nhopself", "value": "0"}]
> Jan 13 07:12:21.562566 vlab-01 DEBUG bgp#bgpcfgd: Received message : '('10.0.0.57', 'SET', (('asn', '1'), ('holdtime', '180'), ('keepalive', '60'), ('local_addr', '10.0.0.58'), ('name', 'BGPMonitor'), ('nhopself', '0')))'
> Jan 13 07:12:21.562566 vlab-01 ERR bgp#bgpcfgd: Peer '(default|10.0.0.57)': Can't update the peer. Only 'admin_status' attribute is supported
> Jan 13 07:12:21.898524 vlab-01 NOTICE config: Patch Applier:   * [{"op": "add", "path": "/BGP_MONITORS/10.0.0.57/rrclient", "value": "0"}]
> Jan 13 07:12:22.256744 vlab-01 DEBUG bgp#bgpcfgd: Received message : '('10.0.0.57', 'SET', (('asn', '1'), ('holdtime', '180'), ('keepalive', '60'), ('local_addr', '10.0.0.58'), ('name', 'BGPMonitor'), ('nhopself', '0'), ('rrclient', '0')))'
> Jan 13 07:12:22.256744 vlab-01 ERR bgp#bgpcfgd: Peer '(default|10.0.0.57)': Can't update the peer. Only 'admin_status' attribute is supported
> Jan 13 07:12:22.625071 vlab-01 NOTICE config: Patch Applier: Verifying patch updates are reflected on ConfigDB.
> Jan 13 07:12:23.670024 vlab-01 NOTICE config: Patch Applier: Patch application completed.
> ```

Any insight for bgpcfgd issue?

**Comment by ganglyu** [comment]
> 

https://github.com/Azure/sonic-buildimage/blob/master/src/sonic-yang-models/yang-models/sonic-bgp-neighbor.yang#L64-L69

File: src/sonic-yang-models/yang-models/sonic-bgp-monitor.yang (hunk 2 above)
Comment (qiluo-msft) [reply src/sonic-yang-models/yang-models/sonic-bgp-monitor.yang:29]: > This table fields seem to be duplicate from the table BGP_NEIGHBOR, please clarify why do we need a separate table

This is a good question, and I believe it is large than the PR trying to solve, ie, add the missing yang model for BGP_MONITORS table, which is existing for long,.

This PR will help minigraph parser unit test improvement, which validate each minigraph generated ConfigDB by SONiC Yang model.

This PR will also unblock the generic config updater end-to-end test.

We can create an issue that BGP_NEIGHBOR and BGP_MONITORS tables should be unified.

Comment (venkatmahalingam) [reply src/sonic-yang-models/yang-models/sonic-bgp-monitor.yang:33]: Sure.

File: src/sonic-yang-models/tests/yang_model_tests/tests_config/bgp.json [hunk 4]
```diff
@@ -1322,8 +1322,87 @@
...
+        "sonic-bgp-monitor:sonic-bgp-monitor": {
+            "sonic-bgp-monitor:BGP_MONITORS": {
+                "BGP_MONITORS_LIST": [
+                    {
+                        "addr": "10.0.0.1",
+                        "asn": "65200",
+                        "holdtime": "180",
+                        "keepalive": "60",
+                        "local_addr": "10.0.0.2",
+                        "name":"BGPMonitor123",
```
Comment (venkatmahalingam) [inline src/sonic-yang-models/tests/yang_model_tests/tests_config/bgp.json:1377]: Does this test case validate 'name' field? having the digit in the name is invalid?

Comment (ganglyu) [reply src/sonic-yang-models/tests/yang_model_tests/tests_config/bgp.json:1377]: https://github.com/Azure/sonic-buildimage/blob/master/src/sonic-config-engine/minigraph.py#L823
It seems that the name must be 'BGPMonitor'

**Review by venkatmahalingam — APPROVED** [review]

File: src/sonic-yang-models/yang-models/sonic-bgp-monitor.yang (hunk 2 above)
Comment (qiluo-msft) [reply src/sonic-yang-models/yang-models/sonic-bgp-monitor.yang:33]: Created issue https://github.com/Azure/sonic-buildimage/issues/9793

**Review by qiluo-msft — APPROVED** [review]

**Comment by ghooo** [comment]
> > Tested on KVM env. `apply-patch` can succeed. But bgpcfgd report ERR. It may be caused by invalid value in my patch because I am not sure how to configure that.
> > BTW, I saw in production that `asn` could be 0.
> > ```
> > admin@vlab-01:~/bgpl$ cat add.json
> > [ {
> >   "op": "add",
> >   "path": "/BGP_MONITORS",
> >   "value": {
> >    "10.0.0.57": {
> >     "asn": "1",
> >     "holdtime": "180",
> >     "keepalive": "60",
> >     "local_addr": "10.0.0.58",
> >     "name": "BGPMonitor",
> >     "nhopself": "0",
> >     "rrclient": "0"
> >    }
> >   }
> >  }
> > ]
> > 
> > admin@vlab-01:~/bgpl$ sudo cat /var/log/syslog | less
> > 
> > Jan 13 07:12:17.930661 vlab-01 NOTICE config: Patch Applier: Applying 7 changes in order:
> > Jan 13 07:12:17.930783 vlab-01 NOTICE config: Patch Applier:   * [{"op": "add", "path": "/BGP_MONITORS", "value": {"10.0.0.57": {"asn": "1"}}}]
> > Jan 13 07:12:18.269504 vlab-01 DEBUG bgp#bgpcfgd: Received message : '('10.0.0.57', 'SET', (('asn', '1'),))'
> > Jan 13 07:12:18.271356 vlab-01 WARNING bgp#bgpcfgd: Peer 10.0.0.57. Missing attribute 'local_addr'
> > Jan 13 07:12:18.272549 vlab-01 INFO bgp#bgpcfgd: Routing policy for peer '10.0.0.57' has been scheduled to be updated
> > Jan 13 07:12:18.272549 vlab-01 INFO bgp#bgpcfgd: Peer-group for peer '10.0.0.57' has been scheduled to be updated
> > Jan 13 07:12:18.273635 vlab-01 INFO bgp#bgpcfgd: Peer '(default|10.0.0.57)' has been scheduled to be added with attributes '{'asn': '1'}'
> > Jan 13 07:12:18.274026 vlab-01 DEBUG bgp#bgpcfgd: execute command '['vtysh', '-f', '/tmp/tmph_geju49']'.
> > Jan 13 07:12:18.399504 vlab-01 ERR bgp#bgpcfgd: command execution returned 2. Command: '['vtysh', '-f', '/tmp/tmph_geju49']', stdout: '', stderr: 'line 29: % Unknown command[24]:   neighbor 10.0.0.57 description #012'
> > Jan 13 07:12:18.400561 vlab-01 ERR bgp#bgpcfgd: ConfigMgr::commit(): can't push configuration from file='/tmp/tmph_geju49', rc='2', stdout='', stderr='line 29: % Unknown command[24]:   neighbor 10.0.0.57 description #012'
> > Jan 13 07:12:18.401146 vlab-01 CRIT bgp#bgpcfgd: Runner::commit was unsuccessful
> > Jan 13 07:12:18.574125 vlab-01 NOTICE config: Patch Applier:   * [{"op": "add", "path": "/BGP_MONITORS/10.0.0.57/holdtime", "value": "180"}]
> > Jan 13 07:12:18.926387 vlab-01 DEBUG bgp#bgpcfgd: Received message : '('10.0.0.57', 'SET', (('asn', '1'), ('holdtime', '180')))'
> > Jan 13 07:12:18.926801 vlab-01 ERR bgp#bgpcfgd: Peer '(default|10.0.0.57)': Can't update the peer. Only 'admin_status' attribute is supported
> > Jan 13 07:12:19.269490 vlab-01 NOTICE config: Patch Applier:   * [{"op": "add", "path": "/BGP_MONITORS/10.0.0.57/keepalive", "value": "60"}]
> > Jan 13 07:12:19.599301 vlab-01 DEBUG bgp#bgpcfgd: Received message : '('10.0.0.57', 'SET', (('asn', '1'), ('holdtime', '180'), ('keepalive', '60')))'
> > Jan 13 07:12:19.599526 vlab-01 ERR bgp#bgpcfgd: Peer '(default|10.0.0.57)': Can't update the peer. Only 'admin_status' attribute is supported
> > Jan 13 07:12:19.926194 vlab-01 NOTICE config: Patch Applier:   * [{"op": "add", "path": "/BGP_MONITORS/10.0.0.57/local_addr", "value": "10.0.0.58"}]
> > Jan 13 07:12:20.264585 vlab-01 DEBUG bgp#bgpcfgd: Received message : '('10.0.0.57', 'SET', (('asn', '1'), ('holdtime', '180'), ('keepalive', '60'), ('local_addr', '10.0.0.58')))'
> > Jan 13 07:12:20.264926 vlab-01 ERR bgp#bgpcfgd: Peer '(default|10.0.0.57)': Can't update the peer. Only 'admin_status' attribute is supported
> > Jan 13 07:12:20.574737 vlab-01 NOTICE config: Patch Applier:   * [{"op": "add", "path": "/BGP_MONITORS/10.0.0.57/name", "value": "BGPMonitor"}]
> > Jan 13 07:12:20.896012 vlab-01 DEBUG bgp#bgpcfgd: Received message : '('10.0.0.57', 'SET', (('asn', '1'), ('holdtime', '180'), ('keepalive', '60'), ('local_addr', '10.0.0.58'), ('name', 'BGPMonitor')))'
> > Jan 13 07:12:20.896012 vlab-01 ERR bgp#bgpcfgd: Peer '(default|10.0.0.57)': Can't update the peer. Only 'admin_status' attribute is supported
> > Jan 13 07:12:21.228961 vlab-01 NOTICE config: Patch Applier:   * [{"op": "add", "path": "/BGP_MONITORS/10.0.0.57/nhopself", "value": "0"}]
> > Jan 13 07:12:21.562566 vlab-01 DEBUG bgp#bgpcfgd: Received message : '('10.0.0.57', 'SET', (('asn', '1'), ('holdtime', '180'), ('keepalive', '60'), ('local_addr', '10.0.0.58'), ('name', 'BGPMonitor'), ('nhopself', '0')))'
> > Jan 13 07:12:21.562566 vlab-01 ERR bgp#bgpcfgd: Peer '(default|10.0.0.57)': Can't update the peer. Only 'admin_status' attribute is supported
> > Jan 13 07:12:21.898524 vlab-01 NOTICE config: Patch Applier:   * [{"op": "add", "path": "/BGP_MONITORS/10.0.0.57/rrclient", "value": "0"}]
> > Jan 13 07:12:22.256744 vlab-01 DEBUG bgp#bgpcfgd: Received message : '('10.0.0.57', 'SET', (('asn', '1'), ('holdtime', '180'), ('keepalive', '60'), ('local_addr', '10.0.0.58'), ('name', 'BGPMonitor'), ('nhopself', '0'), ('rrclient', '0')))'
> > Jan 13 07:12:22.256744 vlab-01 ERR bgp#bgpcfgd: Peer '(default|10.0.0.57)': Can't update the peer. Only 'admin_status' attribute is supported
> > Jan 13 07:12:22.625071 vlab-01 NOTICE config: Patch Applier: Verifying patch updates are reflected on ConfigDB.
> > Jan 13 07:12:23.670024 vlab-01 NOTICE config: Patch Applier: Patch application completed.
> > ```
> 
> Any insight for bgpcfgd issue?

I think this is because the fields under BGP_MONITORS are all create-only. The patch steps are adding each field one by one, this is not correct, The field needs to be added when the parent is created, if they are to be updated The parent needs to be deleted and added back. related issue: https://github.com/Azure/sonic-utilities/issues/2029

**Comment by wen587** [comment]
Hi Mohamed, I agree with you on the create-only conclusion. It is not an urgent need to fix as it can work now after I modify the patch.
Above example are trying to update peer address 10.0.0.57 which may not be a valid operation.
Check the test example here: 
https://github.com/Azure/sonic-mgmt/pull/4987

**Pull request merged**